ansible-scylla-node/tasks/upgrade/upgrade_redhat.yml

```diff
@@ -0,0 +1,49 @@
+---
+
+# Remove existing Scylla repositories
+- name: Remove existing Scylla repositories
+  ansible.builtin.file:
+    path: /etc/yum.repos.d/scylla.repo
+    state: absent    
+  notify: Clean yum metadata  
+  become: true
+  when: upgrade_major
+
+
+# Create Scylla repositories file
+- name: Create Scylla repositories file
+  ansible.builtin.get_url:
+    url: "http://downloads.scylladb.com/rpm/centos/scylla-{{ scylla_upgrade['major_version'] }}.repo"
+    dest: /etc/yum.repos.d/scylla.repo
```

Comment:
make sure this is in sync with what was used in nodes_params , resp. default install for role (see field cloud)

Reply:
This was done on purpose since you don't need to set the repo: it will be set automatically based on your OS, Scylla edition and version. It will use the same destination file each time, avoiding more than one Scylla repo configured in the node.

If this mechanism is not used, you won't be able to use automation for upgrading since each major version has its own repo. This update is taking care about that so the user doesn't need to be worried about it.

IMO, same logic should be applied to the "install" feature (aka the original role).